openedx/edx-ora2
# feat: selectable students waiting step

### Description
This PR adds a toggable feature that allows instructors to automatically search a student after selecting it from the ORA waiting list table:

![ora-update-demo](https://github.com/openedx/edx-ora2/assets/134975835/4f84def3-7fe2-4d24-9b6f-ac13da97f662)



Here's what you see in the video explained:
- There's a student in the ORA waiting list after not receiving an assessment from their peers
- The instructor selects the Open Responses tab from the instructor dashboard
- The table shows the selectable column
- The instructor selects the number of students waiting for an assessment
- The instructor selects the student from the table
- A `Search` button appears
- The instructor presses the search button
- The information on the student's submission is shown
Now the instructor can grade the student.

Why are we proposing these changes? This improvement makes it easier for the instructors to grade ORA submissions without copying and pasting each username for a manual search. As said before, this feature can be turned on and off in case you want the default ORA behavior.

### How to test

- Create at least two students.
- Go to your course -> Create a new unit -> Create an open response component -> `Peer Assessment`.
- Then edit the open response -> Edit -> Assessment Steps.
      -> Select Step Peer Assessment option.
           `Must Grade`: 1
          `Grade By`: 1
- As students upload the open responses:
- Have student 1 fill out the open response and submit.
- Have student 2 fill out the open response, submit, and check student 1's open response.
- As an `instructor`, go to the Instructor panel ->` Open Responses `-> `Waiting` column.
- As an `instructor`, you can see the list of students. Select the one that you want to review and click on the `Search Learner` button.
     

#### DETAILED INSTRUCTIONS, WITH CONFIGURATIONS #####
You must enable these two features
```
FEATURES["ENABLE_XBLOCK_VIEW_ENDPOINT"] = True
FEATURES["ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW"] = True
```

Here an example to add them in tutor:

```
from tutor import hooks

hooks.Filters.ENV_PATCHES.add_items(
    [
        (
            "openedx-lms-development-settings",
            """
FEATURES["ENABLE_XBLOCK_VIEW_ENDPOINT"] = True
FEATURES["ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW"] = True

"""
        ),
    ]
)

```

**Developer Checklist**


**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
